efi/secureboot_namespace_rules_test.go

```diff
@@ -0,0 +1,209 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package efi_test
+
+import (
+	"crypto/x509"
+	"errors"
+
+	. "gopkg.in/check.v1"
+
+	. "github.com/snapcore/secboot/efi"
+	"github.com/snapcore/secboot/internal/efitest"
+	"github.com/snapcore/secboot/internal/testutil"
+)
+
+type secureBootNamespaceRulesSuite struct{}
+
+var _ = Suite(&secureBootNamespaceRulesSuite{})
+
+func (s *secureBootNamespaceRulesSuite) TestRulesMatch1(c *C) {
```

Comment:
this Rules# tests might need some comments to explain what is tested in each, just skimming them is hard to spot the differences between each

Reply:
I've added some comments here now.